pkg/sql/schemachanger/scbuild/build.go

```diff
@@ -82,13 +84,13 @@ func Build(
 			// cluster version.
 			continue
 		}
-		if e.previous == scpb.InvalidTarget {
+		if e.metadata.StatementID == currentStatementID {
 			// The target was newly-defined by this build.
 			if e.target.Status() == e.current {
 				// Discard it if it's already fulfilled.
 				continue
 			}
-		} else if e.previous.InitialStatus() == scpb.Status_ABSENT && e.target == scpb.ToAbsent {
+		} else if e.initial == scpb.Status_ABSENT && e.target == scpb.ToAbsent {
 			// The target was defined as an adding target by a previous build.
 			// This build redefines it as a dropping target.
 			// As far as the schema change is concerned, this is a no-op, so we
```

Comment:
Is there anything to update in this TODO?

Reply:
Ouch. The whole if-block should be superfluous at this point. I just forgot to remove it. Hopefully that's indeed the case, let's find out.

Reply:
It is, phew.